contrib/pkg/adm/managedns/enable.go

```diff
@@ -165,7 +174,13 @@ func (o *Options) Run(dynClient client.Client, args []string) error {
 
 	// Adding manageDNS to HiveConfig triggers a new rollout of the hiveadmission pods.
 	// To know when it's safe to proceed, we will list all hiveadmission pods and ensure
-	// they all have the expected env var set.
+	// they all reference the new configmap.
+	log.Debug("getting new configmap with updated dns data")
+	dnsConfigMapName, err := getUpdatedManagedDNSConfig(dynClient, currHiveadmissionGeneration)
```

Comment:
The hiveutil command will return an error if the managed DNS is unchanged because it will time out waiting for an update to the generation of the hiveadmission deployment.